assignments/array-methods.js

```diff
@@ -2,82 +2,120 @@
+    
+    // Problem 1
+    // Generate an array of company names with their donation amount so that the event director can create sponsorship material. Sort the array by donation amount. 
+    let sponsors = [];
+    runners.forEach(function(runner) {
+        sponsors.push({"company_name": runner.company_name, "donation": runner.donation});
+    });
+    console.log(sponsors.sort(function(a, b) {
+        return a.donation - b.donation;
+    }));
 
-// ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
-
-// ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
-
-// ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
-
-// ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
-
-// ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
-
-// Problem 1
-
-// Problem 2
-
-// Problem 3
\ No newline at end of file
+    // Problem 2
+    // Oh no! The venue for the run has been washed out, and a new, smaller venue has been procured. The runners will have to be split into two groups to accomodate the run. Output the runners whose last names start with a-m to a new array.
+    let runnerGroupOne = runners.filter(function(runner) {
+        return /^[A-Ma-m]/.test(runner.last_name);
+    });
+    console.log(runnerGroupOne);
+    
+    // Problem 3
+    // The event director would like to make a plaque of the companies who donated at least $200 to the center. Generate a new array containing of the companies who donated at least $200 and capitalize the company names.
+    let spendFil = runners.filter(function(runner) {
+        return runner.donation >= 200;
+    });
+    let spendMap = spendFil.map(function(runner) {
+        return runner.company_name.toUpperCase();
+    });
```

Comment:
Try chaining these methods together for practice! It's less code, though ultimately it's a personal choice.